diff --git a/Makefile b/Makefile
index 3f67bbf..e4a1adc 100644
--- a/Makefile
+++ b/Makefile
@@ -67,7 +67,7 @@ TARFILES := AUTHORS BUGS NEWS README TODO COPYING COPYING.LIB \
 # plus the top-level Makefile to make it work stand-alone.
 _TARFILES := Makefile
 
-CURSES := -lncursesw
+CURSES := -ltinfo -lncursesw
 
 # This seems about right for the dynamic library stuff.
 # Something like this is probably needed to make the SE Linux
